proposals/lip-0058.md

```diff
@@ -13,58 +13,66 @@ Requires: 0055, 0056, 0061
 
 ## Abstract
 
-This LIP introduces the BFT module, which is responsible for maintaining the consensus participants, their BFT weights and all information related to the consensus votes that have been cast as part of the block headers. In this LIP, we specify how this information is serialized and stored as well as updated as part of the block processing. Additionally, we specify the functions that can be called from other modules and off-chain services.
+This LIP defines the BFT store and the block processing logic related to the Lisk-BFT consensus protocol. The BFT store is responsible for maintaining the consensus participants, their BFT weights and all information related to the consensus votes that have been cast as part of the block headers. In this LIP, we specify how this information is serialized and stored as well as updated as part of the block processing.
 
 ## Copyright
 
 This LIP is licensed under the [Creative Commons Zero 1.0 Universal](https://creativecommons.org/publicdomain/zero/1.0/).
 
 ## Motivation
 
-The Lisk-BFT consensus protocol introduced in [LIP 0014](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0014.md) specifies two new block properties, `maxHeightGenerated` (renamed from `maxHeightPreviouslyForged`) and `maxHeightPrevoted`, which are validated as part of the block processing. As part of the new state architecture and stricter separation between state machine layer and replication layer, the majority of the block verification now happens in the state machine layer and is logically separated into separate modules. That is why the consensus participants, their BFT weights and the current state of consensus votes is also maintained as part of the state machine. We therefore propose to introduce the BFT module that defines how this information is stored and updated as part of the block processing.
+The Lisk-BFT consensus protocol introduced in [LIP 0014](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0014.md) specifies two new block properties, `maxHeightGenerated` (renamed from `maxHeightPreviouslyForged`) and `maxHeightPrevoted`, which are validated as part of the block processing. As part of the [new state architecture][lip-0069] and stricter separation between state machine layer and replication layer, the majority of the block processing now happens in the state machine layer and is logically separated into separate modules. The consensus-related logic and verification of the block header properties is performed in the replication layer, however. That is why the consensus participants, their BFT weights, keys and the current state of consensus votes is also maintained as part of the replication layer. In this LIP we therefore define the BFT store for this purpose and specify how this information is updated as part of the block processing. Additionally, we define the concept of block slots and the logic for determining which validator is assigned to a specific block slot.
```

Comment:
I would also mention the property `impliesMaximalPrevotes` (although it is not introduced in LIP14), since it is handled by the BFT store